lib/src/obs/mod.rs

```diff
@@ -0,0 +1,94 @@
+//! This module defines the operations for object storage using the [object_store](https://docs.rs/object_store/latest/object_store/)
+//! crate. This will enable the user to store objects using local file storage, or cloud storage such as S3 or GCS.
+use crate::err::Error;
+use bytes::Bytes;
+use futures::stream::BoxStream;
+use object_store::local::LocalFileSystem;
+use object_store::parse_url;
+use object_store::path::Path;
+use object_store::ObjectStore;
+use once_cell::sync::Lazy;
+use sha1::{Digest, Sha1};
+use std::env;
+use std::fs;
+use std::sync::Arc;
+use url::Url;
+
+static STORE: Lazy<Arc<dyn ObjectStore>> =
+	Lazy::new(|| match std::env::var("SURREAL_OBJECT_STORE") {
+		Ok(url) => {
+			let url = Url::parse(&url).expect("Expected a valid url for SURREAL_OBJECT_STORE");
+			let (store, _) =
+				parse_url(&url).expect("Expected a valid url for SURREAL_OBJECT_STORE");
+			Arc::new(store)
+		}
+		Err(_) => {
+			let path = env::current_dir().unwrap().join("store");
+			if !path.exists() || !path.is_dir() {
+				fs::create_dir_all(&path)
+					.expect("Unable to create directory structure for SURREAL_OBJECT_STORE");
+			}
+			// As long as the provided path is correct, the following should never panic
+			Arc::new(LocalFileSystem::new_with_prefix(path).unwrap())
+		}
+	});
+
+static CACHE: Lazy<Arc<dyn ObjectStore>> =
+	Lazy::new(|| match std::env::var("SURREAL_OBJECT_CACHE") {
+		Ok(url) => {
+			let url = Url::parse(&url).expect("Expected a valid url for SURREAL_OBJECT_CACHE");
+			let (store, _) =
+				parse_url(&url).expect("Expected a valid url for SURREAL_OBJECT_CACHE");
+			Arc::new(store)
+		}
+		Err(_) => {
+			let path = env::current_dir().unwrap().join("cache");
+			if !path.exists() || !path.is_dir() {
+				fs::create_dir_all(&path)
+					.expect("Unable to create directory structure for SURREAL_OBJECT_CACHE");
+			}
+			// As long as the provided path is correct, the following should never panic
+			Arc::new(LocalFileSystem::new_with_prefix(path).unwrap())
+		}
+	});
+
+/// Gets the file from the local file system object storage.
+pub async fn stream(
+	file: String,
+) -> Result<BoxStream<'static, Result<Bytes, object_store::Error>>, Error> {
+	match CACHE.get(&Path::from(file.as_str())).await {
+		Ok(data) => Ok(data.into_stream()),
+		_ => Ok(STORE.get(&Path::from(file.as_str())).await?.into_stream()),
+	}
+}
+
+/// Gets the file from the local file system object storage.
+pub async fn get(file: &str) -> Result<Vec<u8>, Error> {
```

Comment:
(Reposting https://github.com/surrealdb/surrealdb/pull/3015)

Are we agreed on exposing these functions now and maintaining backward compatibility until v2?

Reply:
These are exposed publicly, but are hidden from our documentation, and therefore do not need to adhere to our backward compatibility guarantees. We do need to document this somewhere though!